tests/5.1/begin_end_declare_variant.c

```diff
@@ -0,0 +1,76 @@
+//===---- begin_end_declare_variant.c ---------------------------------------------===//
+// 
+// OpenMP API Version 5.1
+//
+// Uses 'begin' and 'end' declare variant syntax to create a parallel and target
+// variant of a simple add function. To ensure that the variants are used, different
+// integers are used within the variants.
+// 
+//===-------------------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+int arr[N]; // implicit map array 
+int errors;
+int i = 0;
+
+void add(int *arr){
+	for (int i = 0; i < N; i++){ // Base function adds 1 to array values
+		arr[i] = i+1;
+	}
+}
+
+#pragma omp begin declare variant match( construct={parallel} ) // Parallel variant adds 2 to array values
+	void add(int *arr){
+		#pragma omp for
+		for (int i = 0; i < N; i++){
+			arr[i] = i + 2;
```

Comment:
add a check in this region using omp_is_initial_device() to make sure that is running on host and not the target